src/t_stream.c

```diff
@@ -2029,30 +2029,56 @@ void xackCommand(client *c) {
     if (ids != static_ids) zfree(ids);
 }
 
-/* XPENDING <key> <group> [<start> <stop> <count> [<consumer>]]
+/* XPENDING <key> <group> [<start> <stop> <count> [<consumer>] [IDLE <idle>]]
  *
  * If start and stop are omitted, the command just outputs information about
  * the amount of pending messages for the key/group pair, together with
  * the minimum and maximum ID of pending messages.
  *
  * If start and stop are provided instead, the pending messages are returned
- * with informations about the current owner, number of deliveries and last
+ * with information about the current owner, number of deliveries and last
  * delivery time and so forth. */
 void xpendingCommand(client *c) {
     int justinfo = c->argc == 3; /* Without the range just outputs general
                                     informations about the PEL. */
     robj *key = c->argv[1];
     robj *groupname = c->argv[2];
-    robj *consumername = (c->argc == 7) ? c->argv[6] : NULL;
+    robj *consumername = NULL;
     streamID startid, endid;
     long long count;
+    long long minidle = 0;
+
+    /* Must contain at least key and group. */
+    if (c->argc < 3) {
+        addReply(c,shared.syntaxerr);
+        return;
+    }
 
-    /* Start and stop, and the consumer, can be omitted. */
-    if (c->argc != 3 && c->argc != 6 && c->argc != 7) {
+    /* Start and stop, and the consumer, can be omitted. Also the IDLE modifier. */
+    if (c->argc < 6 || c->argc > 9) {
```

Comment:
valid values are [3,6,7,8,9]
```suggestion
    if (c != 3 && (c->argc < 6 || c->argc > 9)) {
```
this makes it clear that the case of just 3 arguments isn't covered by the tests